examples/flutter_gallery/lib/demo/expansion_panels_demo.dart

```diff
@@ -284,22 +291,27 @@ class _ExpansionPanelsDemoState extends State<ExpasionPanelsDemo> {
             });
           }
 
-          return new CollapsibleBody(
-            child: new Slider(
-              value: item.value,
-              min: 0.0,
-              max: 100.0,
-              divisions: 5,
-              activeColor: Colors.orange[100 + (item.value * 5.0).round()],
-              label: '${item.value.round()}',
-              onChanged: (double value) {
-                setState(() {
-                  item.value = value;
-                });
-              }
-            ),
-            onSave: close,
-            onCancel: close
+          return new Form(
+            key: form3Key,
+            child: new CollapsibleBody(
+              child: new FormField<double>(
+                initialValue: item.value,
+                onSaved: (double value) { item.value = value; },
+                builder: (FormFieldState<double> field) {
+                  return new Slider(
+                    min: 0.0,
+                    max: 100.0,
+                    divisions: 5,
+                    activeColor: Colors.orange[100 + (field.value * 5.0).round()],
+                    label: '${field.value.round()}',
+                    value: field.value,
+                    onChanged: field.onChanged,
+                  );
+                },
+              ),
+              onSave: () { form3Key.currentState.save(); close(); },
```

Comment:
That could work here where we have a custom widget that implements the onSave function. But sadly there doesn't seem to be a general solution to this issue (see text_field_demo, where we want to call save when a generic RaisedButton is pressed).

I've actually found myself wanting something like this multiple times in this patch - a way for a child widget to refer to its ancestor. The whole reason FormField takes a builder function instead of a child widget is that I want to use FormField's state in the child's builder. A builder function is the only way for child widgets to depend on their ancestor's state in some way.